src/main/kotlin/com/openlattice/users/Auth0SyncTask.kt

```diff
@@ -22,25 +22,31 @@
 package com.openlattice.users
 
 
+import com.google.common.base.Stopwatch
 import com.openlattice.tasks.HazelcastFixedRateTask
 import com.openlattice.tasks.HazelcastTaskDependencies
 import com.openlattice.tasks.Task
 import org.slf4j.LoggerFactory
+import java.time.Instant
 import java.util.concurrent.Semaphore
 import java.util.concurrent.TimeUnit
 
-const val REFRESH_INTERVAL_MILLIS = 120_000L
-private const val DEFAULT_PAGE_SIZE = 100
+const val REFRESH_INTERVAL_MILLIS = 120_000L // 2 min
 private const val MAX_JOBS = 8
 private val logger = LoggerFactory.getLogger(Auth0SyncTask::class.java)
 
 /**
  * This is the auth0 synchronization task that runs every REFRESH_INTERVAL_MILLIS in Hazelcast. It requires that
  * syncDependencies be initialized within the same JVM in order to function properly.
- *
  */
 class Auth0SyncTask : HazelcastFixedRateTask<Auth0SyncTaskDependencies>, HazelcastTaskDependencies {
     private val syncSemaphore = Semaphore(MAX_JOBS)
+
+    private var initialized = false
```

Comment:
I don't think having local state works. The task has to be able to run on any of several conductors, and if the one conductor where the task has run ever restarts, it'll never run again.

This data needs to be stored in some global state. I'm looking into Hazelcast's StatefulTask but I'm not sure it'll work with accessing state outside the runTask method (i.e. in the initializeUsers method).

Reply:
Nice catch, you're right!